_posts/2023-09-20-intro-to-the-various-java-based-k8s-oc-tools.adoc

```diff
@@ -0,0 +1,607 @@
+                    </layers>
+                    <filename>${warName}.war</filename>
+                </configuration>
+                <executions>
+                    <execution>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>package</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>com.google.cloud.tools</groupId>
+                <artifactId>jib-maven-plugin</artifactId>
+                <configuration>
+                    <from>
+                        <image>quay.io/wildfly/wildfly-runtime:latest</image>
+                    </from>
+                    <to>
+                        <image>${imageName}:${imageTag}</image>
+                    </to>
+                    <extraDirectories>
+                        <paths>
+                            <path>
+                                <from>target/server</from>
+                                <into>/opt/server</into>
+                            </path>
+                        </paths>
+                        <permissions>
+                            <permission>
+                                <file>/opt/server/**/*</file>
+                                <mode>770</mode>
+                            </permission>
+                        </permissions>
+                    </extraDirectories>
+                    <container>
+                        <user>root</user>
+                    </container>
+                </configuration>
+
+                <executions>
+                    <execution>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>dockerBuild</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+----
+
+Until now we have checked the usages of these tools in several projects.
+
+== Some notes on the implementation of the wildfly-cloud-tests project
+
+The `wildfly-cloud-tests` uses these tools to do complex WildFly based
+tests, so it’s a good material to learn about how to use these tools in
+depth. One internal class that may worth checking is the
+`WildFlyCommonExtension`footnote:[https://github.com/wildfly-extras/wildfly-cloud-tests/blob/main/common/junit-extension/src/main/java/org/wildfly/test/cloud/common/WildFlyCommonExtension.java].
+Here is its class diagram:
+
+image:2023-09-20-k8s/WildFlyCommonExtension.jpg[image]
+
+It works as a JUnit extension that take cares of the image deployment to
+k8s/openshift. For example, it provides methods to deploy/undeploy the
+k8s resources:
+
+[source,java]
+----
+private void startResourcesInList(ExtensionContext context, KubernetesResource kubernetesResource, KubernetesList resourceList) {
```

Comment:
We should be careful about showing these methods IMO. They could change, but it's okay to express the idea if that is the goal.